Formula/traefik.rb

```diff
@@ -0,0 +1,71 @@
+require "language/go"
+
+class Traefik < Formula
+  desc "Træfik, a modern reverse proxy"
+  homepage "https://traefik.io/"
+  url "https://github.com/containous/traefik/releases/download/v1.4.0/traefik-v1.4.0.src.tar.gz"
+  version "1.4.0"
+  sha256 "9c8663796986127c3b348735c1754a1dd2d841317d4739244742681563d6c3e2"
+  head "https://github.com/containous/traefik.git"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  go_resource "github.com/jteeuwen/go-bindata" do
+    url "https://github.com/jteeuwen/go-bindata.git",
+        :revision => "a0ff2567cfb70903282db057e799fd826784d41d"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/containous/traefik").install buildpath.children
+    ENV.prepend_create_path "PATH", buildpath/"bin"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd "src/github.com/jteeuwen/go-bindata/go-bindata" do
+      system "go", "install"
+    end
+
+    cd "src/github.com/containous/traefik" do
+      cd "webui" do
+        system "yarn", "install"
+        system "yarn", "run", "build"
+      end
+      system "go", "generate"
+      system "go", "build", "-o", bin/"traefik", "./cmd/traefik"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    require "socket"
+
+    server = TCPServer.new(0)
+    web_port = server.addr[1]
+    server.close
+
+    server = TCPServer.new(0)
+    http_port = server.addr[1]
```

Comment:
you could potentially end up with the same one twice since you closed the first one already so get both before closing either

Reply:
Right catch, fixed.